consul/src/main/java/com/linecorp/armeria/internal/consul/AgentCheckClient.java

```diff
@@ -0,0 +1,134 @@
+                         final CompletableFuture<Check> future = new CompletableFuture<>();
+                         try {
+                             final Map<String, Check> checks = client.getObjectMapper().readValue(
+                                     response.content().toStringUtf8(),
+                                     client.getObjectMapper().getTypeFactory().constructMapType(
+                                             Map.class, String.class, Check.class));
+                             if (checks == null || checks.isEmpty()) {
+                                 future.complete(null);
+                             } else {
+                                 future.complete(checks.values().stream().findFirst().orElse(null));
+                             }
+                         } catch (JsonProcessingException e) {
+                             future.completeExceptionally(e);
+                         }
+                         return future;
+                     });
+    }
+
+    /**
+     * Get all checks of agent.
+     *
+     * @return a map of checks
+     */
+    public CompletableFuture<Map<String, Check>> checks() {
+        return checks(null, null);
+    }
+
+    /**
+     * Gets a map of checks from agent.
+     * The map's key is concat
+     */
+    public CompletableFuture<Map<String, Check>> checks(
+            @Nullable String serviceName, @Nullable ConsulParameters parameters) {
+        final String serviceFilter = serviceName == null ? "" : "filter=ServiceName==" + serviceName + '&';
+        return client.consulWebClient()
+                     .get("/agent/checks?" + serviceFilter +
+                          (parameters == null ? "" : parameters.toParameterString()))
+                     .aggregate()
+                     .thenApply(response -> {
+                         final ObjectMapper mapper = new ObjectMapper();
+                         try {
+                             return mapper.readValue(response.content().toStringUtf8(),
+                                                     mapper.getTypeFactory().constructMapType(
+                                                             Map.class, String.class, Check.class));
+                         } catch (JsonProcessingException e) {
+                             throw new RuntimeException(e);
+                         }
+                     });
+    }
+
+    /**
+     * Registers a check into the agent.
+     */
+    public CompletableFuture<Void> register(Check check) {
+        final ObjectMapper mapper = new ObjectMapper();
+        try {
+            final String jsonBody = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(check);
+            return client.consulWebClient()
+                         .put("/agent/check/register", jsonBody).aggregate().thenAccept(response -> {});
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Had chat with @trustin, it is better to pass `JsonProcessingException` by `CompletableFuture` 
```suggestion
        try {
            jsonBody = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(check);
        } catch (JsonProcessingException e) {
            final CompletableFuture<Void> completableFuture = new CompletableFuture<>();
            completableFuture.completeExceptionally(e);
            return completableFuture;
        }
        return client.consulWebClient()
                         .put("/agent/check/register", jsonBody).aggregate().thenAccept(response -> {});
```

We could apply this style to this and other API?